File: java-engine/src/main/java/hu/elteik/knowledgelab/javaengine/core/models/rotorrouterdispersion/RotorRouterDispersionRobot.java
package hu.elteik.knowledgelab.javaengine.core.models.rotorrouterdispersion;

import hu.elteik.knowledgelab.javaengine.core.models.base.Color;
import hu.elteik.knowledgelab.javaengine.core.models.base.Robot;
import hu.elteik.knowledgelab.javaengine.core.models.base.RobotState;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class RotorRouterDispersionRobot extends Robot {

    private Long lastUsedEdgeID;

    public RotorRouterDispersionRobot(Long ID, RobotState state, Color color, Long onID, Long destinationID, Long lastUsedEdgeID) {
        super(ID, state, color, onID, destinationID);
        this.lastUsedEdgeID = lastUsedEdgeID;
    }
}

File: dispersion-app/src/app/models/algorithms/rotor-router-dispersion/RotorRouterDispersionRobot.ts
import { Robot } from '../../core/Robot';
import { RobotState } from '../../utils/RobotState';

export class RotorRouterDispersionRobot extends Robot {
  onID: number;
  destinationID: number;

  constructor(
    id: number,
    state: RobotState,
    onID: number,
    destinationID: number
  ) {
    super(id, state);
    this.onID = onID;
    this.destinationID = destinationID;
  }
}
